zcash_primitives/src/sapling/zip32.rs

```diff
@@ -231,12 +229,34 @@ impl DiversifierKey {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum KeyIndex {
+    Master,
+    Child(ChildIndex),
+}
```

Comment:
I'm confused by this type, which doesn't seem to correspond to anything in the ZIP. The path to a key is a sequence of child indices, and so the master key corresponds to the empty path, no?

Reply:
`KeyIndex::Master` corresponds to the middle paragraph of [this section of ZIP 32](https://zips.z.cash/zip-0032#sapling-extended-spending-keys):

> For the master extended spending key, `depth` is 0, `parent_fvk_tag` is 4 zero bytes, and `i` is 0.

Its only purpose is to enable the key corresponding to the path `m` (i.e. the empty path) to be serialized.

Reply:
Got it. I know this is internal, but please add a comment explaining its usage.